packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdk.kt

```diff
@@ -281,6 +287,34 @@ class DdSdk(
         return configBuilder.build()
     }
 
+    private fun buildFirstPartyHosts(
+        firstPartyHosts: List<ReadableMap>
+    ): Map<String, Set<TracingHeaderType>> {
+        /**
+         * Adapts the data format from the React Native SDK configuration to match with the
+         * Android SDK configuration. For example:
+         *
+         * RN config: [{ match: "example.com", propagatorType: [DATADOG, B3] }]
+         * Android config: { "example.com": [DATADOG, B3] }
+         */
+        val firstPartyHostsWithHeaderType = mutableMapOf<String, MutableSet<TracingHeaderType>>()
+
+        firstPartyHosts.forEach {
+            val match = it.getString("match")
+            val propagatorTypes = it.getArray("propagatorTypes")?.asTracingHeaderTypes()
+            if (match != null && propagatorTypes != null && propagatorTypes.isNotEmpty()) {
+                val hostMatch = firstPartyHostsWithHeaderType[match]
+                if (hostMatch != null) {
+                    hostMatch.addAll(propagatorTypes)
+                } else {
+                    firstPartyHostsWithHeaderType[match] = propagatorTypes.toMutableSet()
+                }
+            }
+        }
+
+        return firstPartyHostsWithHeaderType
```

Comment:
nit: probably should be plural at the end firstPartyHostsWithHeaderType`s` to indicate by name that host can have multiple header types.